ee/cron/chunksHostLoads.js

```diff
@@ -0,0 +1,189 @@
+      // Use the last 15 minutes of data.
+      // TODO: make this configurable?
+      StartTime: new Date(now - 1000 * 60 * 15),
+      EndTime: new Date(now),
+      MetricDataQueries: [
+        {
+          Id: PAGE_VIEWS_PER_SECOND,
+          MetricStat: {
+            Metric: {
+              Namespace: 'PrairieLearn',
+              MetricName: 'PageViewsPerSecond',
+              Dimensions: [
+                {
+                  Name: 'ServerGroup',
+                  Value: config.groupName,
+                },
+              ],
+            },
+            Period: 10,
+            Stat: 'Maximum',
+          },
+        },
+        {
+          Id: ACTIVE_WORKERS_PER_SECOND,
+          MetricStat: {
+            Metric: {
+              Namespace: 'PrairieLearn',
+              MetricName: 'CurrentJobs',
+              Dimensions: [
+                {
+                  Name: 'Job Type',
+                  Value: 'python_worker_active',
+                },
+              ],
+            },
+            Period: 10,
+            Stat: 'Maximum',
+          },
+        },
+        {
+          Id: LOAD_BALANCER_REQUESTS_PER_MINUTE,
+          MetricStat: {
+            Metric: {
+              Namespace: 'AWS/ApplicationELB',
+              MetricName: 'RequestCount',
+              Dimensions: [
+                {
+                  Name: 'LoadBalancer',
+                  Value: config.chunksLoadBalancerDimensionName,
+                },
+                {
+                  Name: 'TargetGroup',
+                  Value: config.chunksTargetGroupDimensionName,
+                },
+              ],
+            },
+            Period: 60,
+            Stat: 'Sum',
+          },
+        },
+      ],
+    })
+    .promise();
+
+  const pageViewsPerSecondMetric = metrics.MetricDataResults.find(
+    (m) => m.Id === PAGE_VIEWS_PER_SECOND
+  );
+  const activeWorkersPerSecondMetric = metrics.MetricDataResults.find(
+    (m) => m.Id === ACTIVE_WORKERS_PER_SECOND
+  );
+  const loadBalancerRequestsPerMinuteMetric = metrics.MetricDataResults.find(
+    (m) => m.Id === LOAD_BALANCER_REQUESTS_PER_MINUTE
+  );
+
+  const maxPageViewsPerSecond = Math.ceil(arrayMax(pageViewsPerSecondMetric.Values));
+  const maxActiveWorkersPerSecond = Math.ceil(arrayMax(activeWorkersPerSecondMetric.Values));
+  const maxLoadBalancerRequestsPerMinute = Math.ceil(
+    arrayMax(loadBalancerRequestsPerMinuteMetric.Values)
+  );
+
+  const desiredInstancesByPageViews = maxPageViewsPerSecond / config.chunksPageViewsCapacityFactor;
+  const desiredInstancesByActiveWorkers =
+    maxActiveWorkersPerSecond / config.chunksActiveWorkersCapacityFactor;
+  const desiredInstancesByLoadBalancerRequests =
+    maxLoadBalancerRequestsPerMinute / config.chunksLoadBalancerRequestsCapacityFactor;
+  const desiredInstances = Math.max(
+    desiredInstancesByPageViews,
+    desiredInstancesByActiveWorkers,
+    desiredInstancesByLoadBalancerRequests,
+    1
+  );
```

Comment:
Maybe we want a `ceil()` wrapped around this?

Reply:
I'm going to move the `ceil` to the individual `desiredInstancesBy...` values, so that's most likely unnecessary.